src/python/pants/backend/javascript/nodejs_project_environment.py

```diff
@@ -154,7 +165,14 @@ async def setup_nodejs_project_environment_process(req: NodeJsProjectEnvironment
             append_only_caches=FrozenDict(**per_package_caches, **req.env.project.extra_caches()),
             timeout_seconds=req.timeout_seconds,
             project_digest=project_digest,
-            extra_env=FrozenDict(**req.extra_env, **req.env.project.extra_env()),
+            extra_env=FrozenDict(
+                {
```

Comment:
Added an intermediate dictionary here so duplicate env vars will be merged, the ordering determines priority (last one wins)